14/src/main.c

```diff
@@ -0,0 +1,276 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <sys/time.h>
+
+#define TASK_PART 2
+
+typedef struct {
+    int x;
+    int y;
+} Point;
+
+typedef struct {
+    int min_x;
+    int min_y;
+    int width;
+    int height;
+    char** map;
+} GameMap;
+
+typedef struct {
+    Point* data;
+    int size;
+    int capacity;
+} PointArray;
+
+
+PointArray* create_point_array(int initial_capacity) {
+    PointArray* arr = (PointArray*) malloc(sizeof(PointArray));
+    arr->data = (Point*) malloc(initial_capacity * sizeof(Point));
+    arr->size = 0;
+    arr->capacity = initial_capacity;
+    return arr;
+}
+
+void add_point(PointArray* arr, int x, int y) {
+    if (arr->size == arr->capacity) {
+        arr->capacity *= 2;
+        arr->data = (Point*) realloc(arr->data, arr->capacity * sizeof(Point));
+    }
+    arr->data[arr->size].x = x;
+    arr->data[arr->size].y = y;
+    arr->size++;
+}
+
+void print_map(GameMap* gameMap) {
+    for (int i = 0; i < gameMap->height; i++) {
+        for (int j = 0; j < gameMap->width; j++) {
+            printf("%c", gameMap->map[i][j]);
+        }
+        printf(" %d\n", i);
+    }
+}
+
+
+void draw_line(GameMap* gameMap, Point p1, Point p2) {
+    if (p1.x == p2.x) {
+        // Vertical line
+        int start_y = (p1.y < p2.y) ? p1.y : p2.y;
+        int end_y = (p1.y > p2.y) ? p1.y : p2.y;
+
+        for (int y = start_y; y <= end_y; y++) {
+            gameMap->map[y][p1.x] = '#';
+        }
+    } else if (p1.y == p2.y) {
+        // Horizontal line
+        int start_x = (p1.x < p2.x) ? p1.x : p2.x;
+        int end_x = (p1.x > p2.x) ? p1.x : p2.x;
+
+        for (int x = start_x; x <= end_x; x++) {
+            gameMap->map[p1.y][x] = '#';
+        }
+    }
+    #if TASK_PART == 2
+        for (int x = 0; x < gameMap->width; x++) {
+            gameMap->map[gameMap->height-1][x] = '#';
+        }
+    #endif
+}
```

Comment:
separate do functions

```suggestion
void draw_vertical_line(GameMap* gameMap, Point p1, Point p2) {
    int start_y = (p1.y < p2.y) ? p1.y : p2.y;
    int end_y = (p1.y > p2.y) ? p1.y : p2.y;
    for (int y = start_y; y <= end_y; y++) {
        gameMap->map[y][p1.x] = '#';
    }
}

void draw_horizontal_line(GameMap* gameMap, Point p1, Point p2) {
    int start_x = (p1.x < p2.x) ? p1.x : p2.x;
    int end_x = (p1.x > p2.x) ? p1.x : p2.x;
    for (int x = start_x; x <= end_x; x++) {
        gameMap->map[p1.y][x] = '#';
    }
}

void draw_line(GameMap* gameMap, Point p1, Point p2) {
    if (p1.x == p2.x) {
        draw_vertical_line(gameMap, p1, p2);
    } else if (p1.y == p2.y) {
        draw_horizontal_line(gameMap, p1, p2);
    }
#if TASK_PART == 2
    for (int x = 0; x < gameMap->width; x++) {
        gameMap->map[gameMap->height-1][x] = '#';
    }
#endif
}
}
```